File: go/anagram/anagram.go
package anagram

import (
	"sort"
	"strings"
)

func lcSort(s string) string {
	c := strings.Split(strings.ToLower(s), "")
	sort.Strings(c)
	return strings.Join(c, "")
}

func Detect(s string, candidates []string) (anagrams []string) {

	lc := lcSort(s)
	for _, cand := range candidates {
		lower := strings.ToLower(cand)
		if lcSort(cand) == lc && strings.ToLower(s) != lower {
			anagrams = append(anagrams, cand)
		}
	}
	return

}

File: clojure/anagram/src/anagram.clj
(ns anagram
  (:require [clojure.string :as str]))

(defn anagram-of? [word, cand]
  (let [lc (str/lower-case word)
        lc-cand (str/lower-case cand)]
    (and
      (not= lc lc-cand)
      (= (sort lc) (sort lc-cand)))))

(defn anagrams-for [word prospect-list]
  (filter (partial anagram-of? word) prospect-list))
